src/service_discovery_setup.js

```diff
@@ -0,0 +1,261 @@
+ * depending on the state of the DNSserviceDiscovery manger.
+ *
+ * Returns:
+ *
+ * @param {new_profile object}
+ *          e.g new_profile = { name: "Mozilla IoT Gateway",
+ *                              host: "mozillaGateway",     // Local DNS name
+ *                              txt: {  desc: "descriptive text",
+ *                                      protocols:  "http, https, oauth2 etc..",
+ *                                      power: "default power"
+ *                                    }
+ *                            }
+ */
+DNSserviceDiscovery.prototype.changeProfile = function(new_profile) {
+  try {
+    console.log('Service Discovery is changing profile. The localDomain is ' +
+      'changed to: ${new_profile.host}');
+    this.localName = new_profile.name;
+    this.localDomain = new_profile.host;
+    this.description = new_profile.txt.desc;
+    this.protocol = new_profile.txt.protocols;
+    this.power = new_profile.txt.power;
+
+    const txt = {desc: this.description,
+                 protocols: this.protocol,
+                 power: this.power};
+    const options = {name: this.localName,
+                     host: this.localDomain,
+                     txt: txt};
+
+    this.dnssdHandle = dnssd.Advertisement(dnssd.tcp('http'),
+                                           this.port, options);
+
+    // Check to see if the profile should be active or not.
+    if (this.serviceState) {
+      this.dnssdHandle.start();
+    } else {
+      this.dnssdHandle.stop();
+    }
+  } catch (err) {
+    // We should never get here. Don't attempt to start service discovery
+    // and log an error message. But allow the application to carry on
+    console.error('The service discovery services could not change profile ' +
+      'settings. The error is: ${err}');
+  }
+};
+
+
+/**
+ * A small helper function.
+ *
+ * getmDNSconfig will look at settings that may be in the DB - set by
+ * the user and default settings. It will safely return a profile that
+ * will work for the calling application to use.
+ *
+ * @Returns: Options object
+ *          e.g Options = { name: "Mozilla IoT Gateway",
+ *                         host: "mozillaGateway",     // Local DNS name
+ *                         txt: {  desc: "descriptive text",
+ *                                 protocols:  "http, https, oauth2 etc..",
+ *                                 power: "default power"
+ *                               }
+ *                       }
+ * This function searches /config or db.sqlite3 DB
+ *
+ * @param {}
+ */
+async function getmDNSconfig() {
+  let mDNSserviceDomain;
+
+  try {
+    mDNSserviceDomain = await Settings.get('localDNSname');
+
+    // There is no local service discovery name stored in the DB -
+    // use the default value
+    if (!mDNSserviceDomain) {
+      mDNSserviceDomain = await Settings.get('localDNSname');
+    }
+    const locName = await config.get(
```

Comment:
No need to await `config.get`.

Reply:
Removed. I thought there might be a little delay in reading files from disk? It's removed anyway and has not caused any problems.